wagtail_draftail_anchors/rich_text.py

```diff
@@ -1,11 +1,56 @@
+import importlib
+
+from django.conf import settings
+from django.utils.html import escape
 from draftjs_exporter.dom import DOM
 
+from wagtail import VERSION as wagtail_version
 from wagtail.admin.rich_text.converters.contentstate_models import Block
 from wagtail.admin.rich_text.converters.html_to_contentstate import (
     BlockElementHandler,
     InlineEntityElementHandler,
 )
 
+if wagtail_version >= (3, 0):
+    from wagtail.rich_text import LinkHandler
+else:
+    from wagtail.core.rich_text import LinkHandler
+
+
+# We can't use "anchor", as Wagtail uses this internally for links whose hrefs
+# start with "#"
+ANCHOR_TARGET_IDENTIFIER = "anchor-target"
+
+
+def render_span(attrs):
+    return '<span id="%s">' % escape(attrs["id"])
+
+
+def render_a(attrs):
+    id_ = escape(attrs["id"])
+    return '<a href="#%s" id="%s" data-id="%s">' % (id_, id_, id_)
+
+
+class AnchorIdentifierLinkHandler(LinkHandler):
+    identifier = ANCHOR_TARGET_IDENTIFIER
+
+    @classmethod
+    def get_renderer(cls):
+        renderer = getattr(cls, "_renderer", None)
+        if renderer is None:
+            renderer = getattr(settings, "DRAFTAIL_ANCHORS_RENDERER", render_a)
+            if isinstance(renderer, str):
+                module_path, *attr = renderer.rsplit(".", maxsplit=1)
+                module = importlib.import_module(module_path)
+                renderer = getattr(module, attr[0]) if attr else module
```

Comment:
Could you use https://docs.djangoproject.com/en/4.1/ref/utils/#django.utils.module_loading.import_string here to simplify?